staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/kcp_rest_options_getter.go

```diff
@@ -0,0 +1,37 @@
+package apiserver
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/registry/generic"
+	"k8s.io/klog/v2"
+)
+
+type apiBindingAwareCRDRESTOptionsGetter struct {
+	delegate generic.RESTOptionsGetter
+	crd      *apiextensionsv1.CustomResourceDefinition
+}
+
+func (t apiBindingAwareCRDRESTOptionsGetter) GetRESTOptions(resource schema.GroupResource) (generic.RESTOptions, error) {
+	ret, err := t.delegate.GetRESTOptions(resource)
+	if err != nil {
+		return ret, err
+	}
+
+	if _, bound := t.crd.Annotations["apis.kcp.dev/bound-crd"]; !bound {
+		ret.ResourcePrefix += "/customresources"
+		return ret, nil
+	}
+
+	apiIdentity := t.crd.Annotations["apis.kcp.dev/identity"]
+	if apiIdentity == "" {
+		klog.Errorf("FIXME(ncdc): wildcard list, bound CRD, not using identity!")
+		//return generic.RESTOptions{}, fmt.Errorf("missing 'apis.kcp.dev/identity' annotation")
```

Comment:
is this block still correct?

Reply:
i.e. do we want to klog.Error?

Reply:
I hit a panic during some testing. Could be a test was "bad" and needs updating. I can circle back and see if I can identify the exact scenario.

Reply:
Ok I can undo this and we'll see what happens 😄

Reply:
Ok now I see where I'm getting the panic from. When the dynamic shared informers are doing wildcard list/watch on all GVRs, they're doing a partial metadata request. For partial metadata (for now), we find the first CRD with a matching group+resource. If it happens to find a CRD in system:bound-crds, that will have the `apis.kcp.dev/bound-crd` annotation stored on it (in etcd). But because it's a partial metadata request and we don't have the apiexport virtual workspace in place yet, we don't have any identity associated with this bound CRD.

Options:

1. Keep this logging-but-don't panic approach, knowing it's temporary
2. Exclude bound CRDs from wildcard partial metadata requests that don't have identity specified